setup.sh

```diff
@@ -35,6 +35,17 @@ if [ "$EUID" -eq 0 ]; then
     exit 1
 fi
 
+# Verify user has write permission to this directory
+if [ ! -w . ]; then
+    echo "ERROR: Your user ($USER) does not have write permission to this directory."
+    echo ""
+    ls -ld "$(pwd)"
+    echo ""
+    echo "Please fix file permissions and try again."
+    echo ""
+    exit 1
+fi
+
```

Comment:
I'm combining an enhancement based on #494 to help with installs where the user does not have write access to their local directory.